docs/source/migration.mdx

```diff
@@ -1062,6 +1066,87 @@ So now you can format errors as such:
   },
 ```
 
+#### `error.extensions.exception` is removed
+
+When Apollo Server 3 formats an error, it may add an extension called `exception`. This extension is an object with a field for every *enumerable* property of the originally thrown error. (This does not apply if the originally thrown error was already a `GraphQLError`.) In addition, [when in dev/debug mode](#debug), `exception` contains an array of strings called `stacktrace`.
+
+For example, if this code runs in a resolver:
+
+```js
+const e = new Error("hello");
+e.extraProperty = "bye";
+throw e;
+```
+
+then (in debug mode) Apollo Server 3 will format the error like this:
+
+```js
+{
+  "errors": [
+    {
+      "message": "hello",
+      "locations": [
+        {
+          "line": 2,
+          "column": 3
+        }
+      ],
+      "path": ["x"],
+      "extensions": {
+        "code": "INTERNAL_SERVER_ERROR",
+        "exception": {
+          "extraProperty": "bye",
+          "stacktrace": [
+            "Error: hello",
+            "    at Object.x (file:///private/tmp/as3-t/server.mjs:8:27)",
+            "    at field.resolve (/private/tmp/as3-t/node_modules/apollo-server-core/dist/utils/schemaInstrumentation.js:56:26)",
+            // more lines elided
+          ]
+        }
+      }
+    }
+  ]
+}
+```
+
+It was often hard to predict exactly which properties of which errors would be publicly exposed in this manner, which could lead to surprising information leaks.
+
+In Apollo Server 4, there is no `exception` extension. The `stacktrace` is provided directly on `extensions`. If you'd like to copy some or all properties from the original error onto the formatted error, you can do that with the `formatError` hook.
+
+If you'd like your errors to be formatted like they are in Apollo Server 3 (with the stack trace and the enumerable properties of the original error on the `exception` extension), you can provide this `formatError` implementation:
+
+<MultiCodeBlock>
+
+```ts
+function formatError(
+  formattedError: GraphQLFormattedError,
+  error: unknown,
+) {
+  const exception: Record<string, unknown> = {
+    ...(typeof error === 'object' ? error : null),
```

Comment:
better to do `instanceof Error` check here.
We need to protect against strings and arrays (pass you current check).
<img width="210" alt="image" src="https://user-images.githubusercontent.com/8336157/183117076-3f0ac117-fdb8-4556-9d29-ae69192d763b.png">

Reply:
Also to work with `originalError` it can be
```
...(error instanceof Error ? error.originalError ?? error : null),
```

Reply:
I think I'm not super-worried about things like strings and arrays here... if you pass something ultra weird, then your code ends up weird. I just want to make sure TS builds.